utils/search/search_general.py

```diff
@@ -22,19 +22,320 @@
 import math
 import random
 import socket
+import json
 
+import re
 from django.conf import settings
 
 import sounds
+from search import forms
 from search.forms import SEARCH_SORT_OPTIONS_WEB
-from search.views import search_prepare_sort, search_prepare_query
-from utils.search.solr import Solr, SolrQuery, SolrResponseInterpreter, SolrException
+from utils.search.solr import Solr, SolrQuery, SolrResponseInterpreter, SolrException, \
+    SolrResponseInterpreterPaginator
 from utils.text import remove_control_chars
+from utils.logging_filters import get_client_ip
 
 logger = logging.getLogger("search")
 console_logger = logging.getLogger("console")
 
 
+def search_prepare_sort(sort, options):
+    """ for ordering by rating order by rating, then by number of ratings """
+    if sort in [x[1] for x in options]:
+        if sort == "avg_rating desc":
+            sort = [sort, "num_ratings desc"]
+        elif  sort == "avg_rating asc":
+            sort = [sort, "num_ratings asc"]
+        else:
+            sort = [sort]
+    else:
+        sort = [forms.SEARCH_DEFAULT_SORT]
+    return sort
+
+
+def search_process_filter(filter_query):
+    # Process the filter to replace humnan-readable Audio Commons descriptor names for the dynamic field names used in
+    # Solr (e.g. ac_tonality -> ac_tonality_s, ac_tempo -> ac_tempo_i). The dynamic field names we define in Solr
+    # schema are '*_b' (for bool), '*_d' (for float), '*_i' (for integer) and '*_s' (for string). At indexing time
+    # we append these suffixes to the ac descirptor names so Solr can treat the types properly. Now we automatically
+    # append the suffices to the filter names so users do not need to deal with that.
+    for name, t in settings.AUDIOCOMMONS_INCLUDED_DESCRIPTOR_NAMES_TYPES:
+        filter_query = filter_query.replace('ac_{0}:'.format(name), 'ac_{0}{1}:'
+                                            .format(name, settings.SOLR_DYNAMIC_FIELDS_SUFFIX_MAP[t]))
+    return filter_query
+
+
+def search_prepare_parameters(request):
```

Comment:
This function takes the request as parameter.
This way, the code remains the same as before.
However, it is not very piratical, maybe it would be better to explicitly enter the parameters that are relevant, with default None values for the one that were not present in the url query parameters. 
This would imply some changes in the way things are processed. Basically separating a first step that takes care of getting all the query parameters (request.GET.get()).